src/Engine/GameEngine.cs

```diff
@@ -81,5 +88,55 @@ public void OnError(Exception error)
         {
             throw new NotImplementedException();
         }
+
+        #region IDisposable Implementation
+
+        private bool _disposedValue = false;
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
+            {
+                if (disposing)
+                {
+                    if (_eventSubscriptions != null && _eventSubscriptions.Any())
+                    {
+                        foreach (var item in _eventSubscriptions)
+                        {
+                            foreach (var subscription in item.Value)
+                            {
+                                subscription.Dispose();
+                            }
+
+                            _eventSubscriptions.Remove(item.Key);
+                        }
+                    }
+
+                    if (_timeSubscriptions != null && _timeSubscriptions.Any())
```

Comment:
The code here and the code above for the `_eventSubscriptions` I think is identical bt looking at a different collection. If so, can we remove the duplication with a refactor to method and pass each collection into said method in turn?